### ISSUE TITLE
silverkey fails on linux

### ISSUE THREAD

**Issue by metacoma** [metacoma/silverkey#108]
```
QWidget: MainWindow (skInput) already in focus proxy chain
Exception
Hide action, value is  ""
Reply:  0
*** Error in `/home/bebebeko/Downloads/artifats/archive/src/silverkey-qt-linux-x64_86': free(): invalid pointer: 0x00007ffc1cd04490 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x777e5)[0x7f63bed597e5]
/lib/x86_64-linux-gnu/libc.so.6(+0x8037a)[0x7f63bed6237a]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x4c)[0x7f63bed6653c]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5(_ZN7QObject5eventEP6QEvent+0x100)[0x7f63bfafbce0]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5(_ZN7QWidget5eventEP6QEvent+0x43b)[0x7f63c08a48db]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5(_ZN9QLineEdit5eventEP6QEvent+0x65)[0x7f63c099b075]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x9c)[0x7f63c085ebdc]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5(_ZN12QApplication6notifyEP7QObjectP6QEvent+0x3b0)[0x7f63c08658d0]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0x100)[0x7f63bfad30b0]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x1cb)[0x7f63bfad5aeb]
/home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5(_ZN16QCoreApplication4execEv+0xca)[0x7f63bfad91ba]
/home/bebebeko/Downloads/artifats/archive/src/silverkey-qt-linux-x64_86[0x40e4f0]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7f63bed02830]
/home/bebebeko/Downloads/artifats/archive/src/silverkey-qt-linux-x64_86[0x40ead9]
````

**Comment by metacoma** [comment]
gdb backtrace

```
(gdb) bt
#0  0x00007ffff53e4428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff53e602a in __GI_abort () at abort.c:89
#2  0x00007ffff54267ea in __libc_message (do_abort=do_abort@entry=2, 
    fmt=fmt@entry=0x7ffff553fed8 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff542f37a in malloc_printerr (ar_ptr=<optimized out>, ptr=<optimized out>, 
    str=0x7ffff553ccaf "free(): invalid pointer", action=3) at malloc.c:5006
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3867
#5  0x00007ffff543353c in __GI___libc_free (mem=<optimized out>) at malloc.c:2968
#6  0x00007ffff61c8ce0 in QObject::event(QEvent*) () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5
#7  0x00007ffff6f718db in QWidget::event(QEvent*) () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5
#8  0x00007ffff7068075 in QLineEdit::event(QEvent*) () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5
#9  0x00007ffff6f2bbdc in QApplicationPrivate::notify_helper(QObject*, QEvent*) ()
   from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5
#10 0x00007ffff6f328d0 in QApplication::notify(QObject*, QEvent*) () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Widgets.so.5
#11 0x00007ffff61a00b0 in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5
#12 0x00007ffff61a2aeb in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) ()
   from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5
#13 0x00007ffff61a61ba in QCoreApplication::exec() () from /home/bebebeko/Qt/5.7/gcc_64/lib/libQt5Core.so.5
#14 0x000000000040e4f0 in main ()
```

**Comment by metacoma** [comment]
steps to reproduce:
 1. shutdown etcd backend
 2. run silverkey app

**Comment by metacoma** [comment]
by some reason, this code no create the QmessageBox dialog window, and reply is always 0, tested under ubuntu unity desktop 
```
      int reply = QMessageBox::question(this,
                                      "Error", "Failed to load data from database",
                                      "Quit", "Open Settings");
        qDebug() << "Reply: " << reply;
```

**Comment by metacoma** [comment]
Closed 5076f5c
